strats.md

```diff
@@ -45,3 +47,411 @@ Execution of a strat may have an effect of clearing one or more [obstacles](regi
+
+When a `leaveWithRunway` conditions occurs on a door in a water environment, it implicitly includes a `Gravity` requirement.
+
+#### Example
+```json
+{
+  "name": "Leave With Runway",
+  "notable": false,
+  "requires": [],
+  "exitCondition": {
+    "leaveWithRunway": {
+      "length": 20,
+      "openEnd": 1
+    }
+  }
+}
+```
+
+### Leave Charged
+
+A `leaveCharged` object represents that Samus can leave through this door with a shinecharge (shinespark charge).
+
+`leaveCharged` has a single property:
+- _framesRemaining_: The number of frames remaining in the shinecharge when leaving the room.
+
+A strat with a `leaveCharged` condition should include a `canShinecharge` requirement in its `requires`, as this is not implicitly included in the condition.
+
+*Note*: Using a runway connected to a door to leave the room with a shinecharge is already covered by `leaveWithRunway`, so `leaveCharged` only needs to be used in cases where the shinecharge is obtained in another part of the room and then carried through the door.
+
+#### Example
+```json
+{
+  "name": "Leave Charged",
+  "notable": false,
+  "requires": [
+    {"canShinecharge": {
+      "usedTiles": 20,
+      "openEnd": 0
+    }}
+  ],
+  "exitCondition": {
+    "leaveCharged": {
+      "framesRemaining": 90
+    }
+  }
+}
+```
+
+### Leave With Spark
+
+A `leaveWithSpark` exit condition represents that Samus can leave through this door while shinesparking. A strat with a `leaveWithSpark` condition should include a `canShinecharge` and `shinespark` requirements in its `requires`.
+
+The `leaveWithSpark` object currently has no properties. If needed, properties might be added in the future to describe constraints on the position and direction of the spark. Currently it is implicitly assumed that the position can be fully controlled to be whatever is needed in the next room (e.g. to spark through either the top or bottom part of a horizontal door transition); so the requirements of a `leaveWithSpark` strat should be based on the worst-case scenario. The direction of the spark is assumed to be horizontal when sparking through horizontal door transitions, or vertical when sparking through vertical door transitions.
+
+*Note*: Using a runway connected to a door to leave the room with a shinespark is already covered by `leaveWithRunway`. Likewise `leaveCharged` implicitly includes the possibility of leaving the room with a shinespark. It is only necessary to use `leaveWithSpark` in cases where it would not be possible to reach the door before the shinecharge timer expires.
+
+#### Example
+```json
+{
+  "name": "Leave With Spark",
+  "notable": false,
+  "requires": [
+    {"canShinecharge": {
+      "usedTiles": 20,
+      "openEnd": 0
+    }},
+    {"shinespark": {
+      "frames": 200
```

Comment:
I think it's good to encourage the use of excessFrames
```suggestion
      "frames": 200,
      "excessFrames": 50
```

Reply:
`excessFrames` would mean the comeInWithSpark would fail the health check on the following strat, and that is fine for this strat doubling as a way to reach the door.

But there probably needs to be a field for preopening doors.  Or these can all link to self on the door to show it was opened first.

Reply:
I don't think we want excessFrames on a leaveWithSpark strat. A cross-room strat needs to satisfy the requirements on both sides of the transition or it is not applicable; and if it runs down to 29 energy in the first room then it can't make it to the next room, so the combined strat doesn't apply. So stopping in the middle at the door would not be an option; that would have to be a separate strat.

Reply:
Oh, I was thinking this was the frames post transition. that makes sense